app/ui/client/lib/UserAction.js

```diff
@@ -0,0 +1,177 @@
+import { Meteor } from 'meteor/meteor';
+import { Tracker } from 'meteor/tracker';
+import { ReactiveDict } from 'meteor/reactive-dict';
+import { Session } from 'meteor/session';
+import _ from 'underscore';
+
+import { settings } from '../../../settings';
+import { Notifications } from '../../../notifications';
+
+const timeout = 15000;
+const renew = timeout / 3;
+const rooms = {};
+
+export const USER_RECORDING = 'user-recording';
+export const USER_TYPING = 'user-typing';
+export const USER_UPLOADING = 'user-uploading';
+export const USER_ACTIVITY = 'user-activity';
+
+const recordingTimeouts = {};
+const uploadingTimeouts = {};
+const typingTimeouts = {};
+
+const recordingRenews = {};
+const uploadingRenews = {};
+const typingRenews = {};
+
+const recordingUsers = new ReactiveDict();
+const uploadingUsers = new ReactiveDict();
+const typingUsers = new ReactiveDict();
+
+const selfActivity = { USER_RECORDING: false, USER_UPLOADING: false, USER_TYPING: false };
+
+const shownName = function(user) {
+	if (!user) {
+		return;
+	}
+	if (settings.get('UI_Use_Real_Name')) {
+		return user.name;
+	}
+	return user.username;
+};
+
+const stopActivity = (rid, activityType, extras) => Notifications.notifyRoom(rid, USER_ACTIVITY, shownName(Meteor.user()), false, activityType, extras);
+const startActivity = (rid, activityType, extras) => Notifications.notifyRoom(rid, USER_ACTIVITY, shownName(Meteor.user()), true, activityType, extras);
```

Comment:
Here you are sending an extra parameter, while the backend is dealing and expecting 3 `username, _activity, extraData` you are passing 4 `shownName(Meteor.user()), false, activityType, extras`.

My suggestion is to merge the boolean and activity type as `shownName(Meteor.user()), {type: activityType, activity: false}, extras`

Reply:
true, but it won't give an error right now. We can do it while integrating live chat? I didn't touch it because live chat is using it.